swagger_to/style.py

```diff
@@ -0,0 +1,224 @@
+    """
+    Checks the typedef's adherence to the casing conventions.
+
+    :param typedef: to be translated
+    :param visited: already seen typedefs
+    :return: the list of failed checks
+    """
+    errs = []  # type: List[str]
+    if typedef in visited:
+        return errs
+
+    visited.add(typedef)
+
+    if isinstance(typedef, swagger_to.intermediate.Primitivedef):
+        pass
+
+    elif isinstance(typedef, swagger_to.intermediate.Arraydef):
+        errs.extend(_check_recursively_cases(typedef=typedef.items, visited=visited))
+
+    elif isinstance(typedef, swagger_to.intermediate.Mapdef):
+        errs.extend(_check_recursively_cases(typedef=typedef.values, visited=visited))
+
+    elif isinstance(typedef, swagger_to.intermediate.Objectdef):
+
+        if typedef.identifier != "" and typedef.identifier != swagger_to.capital_camel_case(typedef.identifier):
+            errs.append("{} is not a capital camel case identifier.".format(typedef.identifier))
+
+        for prop in typedef.properties.values():
+            if prop.name != swagger_to.snake_case(prop.name):
+                errs.append("{} is not a snake case identifier.".format(prop.name))
+            errs.extend(_check_recursively_cases(typedef=prop.typedef, visited=visited))
+
+    return errs
+
+
+def _check_recursively_descriptions(typedef: swagger_to.intermediate.Typedef,
+                                    visited: MutableSet[swagger_to.intermediate.Typedef]) -> List[str]:
+    """
+    Checks the typedef's adherence to the description conventions.
+
+    :param typedef: to be translated
+    :param visited: already seen typedefs
+    :return: the list of failed checks
+    """
+    errs = []  # type: List[str]
+    if typedef in visited:
+        return errs
+
+    visited.add(typedef)
+
+    if isinstance(typedef, swagger_to.intermediate.Primitivedef):
+        pass
+
+    elif isinstance(typedef, swagger_to.intermediate.Arraydef):
+        errs.extend(_check_recursively_descriptions(typedef=typedef.items, visited=visited))
+
+    elif isinstance(typedef, swagger_to.intermediate.Mapdef):
+        errs.extend(_check_recursively_descriptions(typedef=typedef.values, visited=visited))
+
+    elif isinstance(typedef, swagger_to.intermediate.Objectdef):
+        if not check_description(typedef.description):
+            errs.append("Description \"{}\" must begin with a lower case and end with a period.".format(
+                typedef.description))
+
+        for prop in typedef.properties.values():
+            if not check_description(prop.description):
+                errs.append("Description \"{}\" must begin with a lower case and end with a period.".format(
+                    prop.description))
+                errs.extend(_check_recursively_descriptions(typedef=prop.typedef, visited=visited))
+
+    return errs
+
+
+def check_endpoint_path_responses(endpoints: List[swagger_to.intermediate.Endpoint]) -> List[str]:
```

Comment:
I'd split these in two even though they both operate on the same input; this makes the final function much clearer and more explicit. If we ever want to add flags like `--skip_endpoint_slash` or something similar, the proposed refactoring would make the implementation fairly easy.